inst/private/doctest_run.m

```diff
@@ -58,41 +65,62 @@
       tests(i).ellipsis = strcmp(directive(1), '+');
     elseif strcmp('+SKIP', directive)
       tests(i).skip = true;
+    elseif strcmp('+SKIP_IF', directive)
+      tests(i).skip = condition;
+    elseif strcmp('+SKIP_UNLESS', directive)
+      tests(i).skip = sprintf('~(%s)', condition);
     elseif strcmp('+XFAIL', directive)
       tests(i).xfail = true;
+    elseif strcmp('+XFAIL_IF', directive)
+      tests(i).xfail = condition;
+    elseif strcmp('+XFAIL_UNLESS', directive)
+      tests(i).xfail = sprintf('~(%s)', condition);
     else
       error('doctest: unexpected directive %s', directive);
     end
   end
 end
 
-% run tests in local namespace
-results = doctest_run_impl(tests);
+% run tests in a local namespace
+results = DOCTEST__run_impl(tests);
 
 end
 
 
 % the following function is used to evaluate all lines of code in same
-% namespace (the one of this invocation of doctest_run_impl)
-function DOCTEST__results = doctest_run_impl(DOCTEST__tests)
+% namespace (the one of this invocation of DOCTEST__run_impl)
+function DOCTEST__results = DOCTEST__run_impl(DOCTEST__tests)
 
 % do not split long rows (TODO: how to do this on MATLAB?)
 if is_octave()
   split_long_rows(0, 'local')
 end
 
+% define test-global constants
+OCTAVE = is_octave;
```

Comment:
That's right. Or even something more explicit like `DOCTEST_OCTAVE`, to avoid polluting the namespace (it would be use as `+SKIP_IF(DOCTEST_OCTAVE)`).